django_mongodb/indexes.py

```diff
@@ -0,0 +1,101 @@
+from django.core.exceptions import EmptyResultSet, FullResultSet
+from django.db import NotSupportedError
+from django.db.models import Index
+from django.db.models.expressions import Col
+from django.db.models.fields.related_lookups import In
+from django.db.models.lookups import BuiltinLookup
+from django.db.models.sql.query import Query
+from django.db.models.sql.where import AND, XOR, WhereNode
+
+from .query_utils import process_rhs
+
+mongo_operators_idx = {
+    "exact": "$eq",
+    "gt": "$gt",
+    "gte": "$gte",
+    "lt": "$lt",
+    "lte": "$lte",
+    "in": "$in",
+}
+
+
+def _get_condition_mql(self, model, schema_editor):
+    """Analogous to Index._get_condition_sql()."""
+    query = Query(model=model, alias_cols=False)
+    where = query.build_where(self.condition)
+    compiler = query.get_compiler(connection=schema_editor.connection)
+    return where.as_mql_idx(compiler, schema_editor.connection)
+
+
+def builtin_lookup_idx(self, compiler, connection):
+    if not isinstance(self.lhs, Col):
+        raise NotSupportedError("Expressions as indexes are not supported.")
```

Comment:
Is this message tested? I don't see it in the new tests or in the Django test suite.

Reply:
Will add.